### ISSUE TITLE
ConnectionRefusedError, Not working 

### ISSUE THREAD

**Issue by Ahmad-Dukhan** [DamnWidget/anaconda#106]
Hi, 

I have this issue with Anaconda, when I set my Venv python path and try to use anaconda (i.e got to definition). Sublime freezes for a while (3-5 min) and then throws this error in the console 

```
[2013-12-15 17:06:28,042 - ERROR - root] Traceback (most recent call last):
  File "/home/ahmed/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/worker.py", line 163, in start
    self.client = AsynClient(self.available_port)
  File "/home/ahmed/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/jsonclient.py", line 34, in __init__
    EventHandler.__init__(self, (host, port))
  File "/home/ahmed/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/ioloop.py", line 105, in __init__
    self.sock.connect(address)
ConnectionRefusedError: [Errno 111] Connection refused
```

I need help please 

P.S Starting sublime as a root didn't help either

Thanks

**Comment by kurtontheway** [comment]
I have the same error message everytime when all plugins loaded, 
plugins loaded
ERROR:root:[Errno 111] Connection refused
ERROR:root:Traceback (most recent call last):
  File "/home/wting/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/worker.py", line 191, in start
    self.client = AsynClient(self.available_port)
  File "/home/wting/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/jsonclient.py", line 36, in __init__
    EventHandler.__init__(self, (host, port))
  File "/home/wting/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/ioloop.py", line 105, in __init__
    self.sock.connect(address)
ConnectionRefusedError: [Errno 111] Connection refused
Registering handler with address ('localhost', 57986)

and this is what I have in the jsonserver.log file:

2015-01-06 12:36:38,669: DEBUG   : bind: address=('localhost', 57986)
2015-01-06 12:36:38,669: DEBUG   : listen: backlog=5
2015-01-06 12:36:38,670: INFO    : Anaconda Server started in port 57986 for PID 14366 with cache dir /home/wting/.cache/jedi/maya2014_lnx and extra paths /apps/shared/maya/2014.4/devkit/other/pymel/extras/completion/py/,/home/wting/kwMayaAppDir/python,/home/wting/kwMayaAppDir/scripts,/home/wting/github/reanimation/src/odwrigging/src/odwrigging/aniPicker,/home/wting/github/reanimation/src/odwrigging/src/odwrigging,/home/wting/kwMayaAppDir/python/kfp3,/home/wting/kwMayaAppDir/python/penguins
2015-01-06 12:36:38,670: INFO    : Incomming connection from ('127.0.0.1', 51682)
2015-01-06 12:36:38,670: INFO    : Incomming connection from ('127.0.0.1', 51683)
2015-01-06 12:36:39,437: ERROR   : Traceback (most recent call last):
  File "/home/wting/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/../anaconda_server/jsonserver.py", line 298, in <module>
    server = JSONServer(('localhost', port))
  File "/home/wting/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/../anaconda_server/jsonserver.py", line 135, in __init__
    self.bind(self.address)
  File "/apps/shared/maya/2014.4/lib/python27.zip/asyncore.py", line 341, in bind
    return self.socket.bind(addr)
  File "/apps/shared/maya/2014.4/lib/python27.zip/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
error: [Errno 98] Address already in use
2015-01-06 12:36:39,437: ERROR   : [Errno 98] Address already in use

**Comment by DamnWidget** [comment]
Hi @kurtontheway 
Is hard to know what is going on, seems like the JsonServer is crashing and trying to start again using the same port than the dead version was used (that is still in pending state by the operating system).

Ahmad solved starting Sublime Text as root (not sure why that solved the issue), is probably that the problem is related with the embedded version of Python being used by Maya3D, but I have no way to test that.

**Comment by kurtontheway** [comment]
Hi @DamnWidget , 
thanks for your reply！
Both autoCompletion and linting are working well here in despite of the error messages. so I'm not sure what's missing.

**Comment by brocksamson** [comment]
I just ran into this issue on my system using virtual environments.  Anaconda was working perfectly for about a week (brand new system) and then suddenly all the linting and autocompletion just stopped functioning.  ps aux showed a number of anaconda json servers running even though I had completely exited out of sublime. I don't remember the exact process name, pretty sure I grepped for python to find them.  After killing them & restarting sublime I noticed that the json servers weren't in my process list anymore.  The sublime console showed the error message from above.

After reading this thread I exited sublime text and rm -rf the .cache/jedi folder and my system is now functioning correctly. 

I have a hunch that the root cause on my system was due to me doing a sudo subl for a config file in /etc.  I suspect that caused some cached files to get written out with the improper owner.  Going forward I plan to use sudo -H subl -n -w whenever I need to edit a root file.  I believe this should avoid the problem.
